pkg/nfd-worker/nfd-worker.go

```diff
@@ -191,34 +227,66 @@ func (w *nfdWorker) Run() error {
 	}
 	defer w.disconnect()
 
+	// Create watcher for config file and read initial configuration
+	configFilePath := filepath.Clean(w.args.ConfigFile)
+	configWatch, paths, err := addConfigWatch(configFilePath)
+	if err != nil {
+		return (err)
+	}
+	w.configure(configFilePath, w.args.Options)
+
+	labelTrigger := time.After(0)
+	var configTrigger <-chan time.Time
 	for {
-		// Parse and apply configuration
-		w.configure(w.args.ConfigFile, w.args.Options)
+		select {
+		case <-labelTrigger:
+			// Get the set of feature labels.
+			labels := createFeatureLabels(w.sources, w.labelWhiteList)
+
+			// Update the node with the feature labels.
+			if w.client != nil {
+				err := advertiseFeatureLabels(w.client, labels)
+				if err != nil {
+					return fmt.Errorf("failed to advertise labels: %s", err.Error())
+				}
+			}
 
-		// Get the set of feature labels.
-		labels := createFeatureLabels(w.sources, w.labelWhiteList)
+			if w.args.Oneshot {
+				return nil
+			}
 
-		// Update the node with the feature labels.
-		if w.client != nil {
-			err := advertiseFeatureLabels(w.client, labels)
-			if err != nil {
-				return fmt.Errorf("failed to advertise labels: %s", err.Error())
+			if w.args.SleepInterval > 0 {
+				labelTrigger = time.After(w.args.SleepInterval)
 			}
-		}
 
-		if w.args.Oneshot {
-			break
-		}
+		case e := <-configWatch.Events:
```

Comment:
I think we should add a log here to specify that we have detected a change in the configuration file

Reply:
At this point we only have detected a change somewhere in the directory tree. E.g. if the config would be `/foo/bar/baz.conf`, any change in any file under `/foo/` or `/foo/bar/` will create an event. We don't want to log all of them, just those concerning `baz.conf` or the directories directly.

But yes, a log message makes sense. I'll add a log a bit later after the `if`

Reply:
Added a log message below